oemof_b3/tools/data_processing.py

```diff
@@ -1,7 +1,734 @@
+        # Convert to datetime format
+        df["timeindex_start"] = pd.to_datetime(df["timeindex_start"])
+        df["timeindex_stop"] = pd.to_datetime(df["timeindex_stop"])
+        # Convert series values from string to list
+        for number, item in enumerate(df["series"].values):
+            df["series"].values[number] = ast.literal_eval(item)
+
+    # "region" can be extraced from var_name. Therefore a further
+    # required header required_header_without_reg is introduced
+    required_header_without_reg = required_header.copy()
+    required_header_without_reg.remove("region")
+    # If optional columns are missing in the stacked DataFrame
+    if (
+        list(df.columns) == required_header
+        or list(df.columns) == required_header_without_reg
+    ) and list(df.columns) != header:
+        # ID in the form of numbering is added if "id_ts" is missing
+        if optional_header[0] not in df.columns:
+            df[optional_header[0]] = np.arange(0, len(df))
+
+        # The region is extracted out of "var_name"
+        if required_header[0] not in df.columns:
+            region = []
+            for row in np.arange(0, len(df)):
+                # "BE_BB" is added if both "BE" and "BB" in var_name
+                if "BE" in df["var_name"][row] and "BB" in df["var_name"][row]:
+                    region.append("BE_BB")
+                # "BE" is added if "BE" in var_name
+                elif "BE" in df["var_name"][row] and "BB" not in df["var_name"][row]:
+                    region.append("BE")
+                # "BB" is added if "BB" in var_name
+                elif "BE" not in df["var_name"][row] and "BB" in df["var_name"][row]:
+                    region.append("BB")
+                # An error is raised since the region is missing in var_name
+                else:
+                    raise ValueError(
+                        "The data is missing the region."
+                        "Please add BB or BE to var_name column"
+                    )
+            # Add list with region to DataFrame
+            df[required_header[0]] = region
+
+        for num_col in np.arange(1, len(optional_header)):
+            # For every other optional column name, an empty array is added with the name as
+            # header - A user info is printed
+            if optional_header[num_col] not in df.columns:
+                df[optional_header[num_col]] = [np.nan] * len(df["series"])
+
+    # Sort the DataFrame to match the header of the template
+    df = df[header]
+
+    return df
+
+
+def save_df(df, path):
+    """
+    This function saves data to a csv file
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        DataFrame to be saved
+
+    path : str
+        Path to save the csv file
+
+    """
+    # Save scalars to csv file
+    df.to_csv(path, index=False)
+
+    # Print user info
+    print(f"User info: The DataFrame has been saved to: {path}.")
+
+
+def df_filtered(df, key, values):
```

Comment:
```suggestion
def filter_df(df, key, values):
```

Reply:
I would name the function by its action, so using the function could look like `df_filtered = filter_df(df)`